Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs

```diff
@@ -189,49 +211,174 @@ public System.Collections.Generic.IEnumerable<string> RedoActionNames
         public RelayCommand ChangeStrokeWidth => changeStrokeWidth;
 
         public RelayCommand ChangeShapeLocation => changeShapeLocation;
+        #endregion
 
-        
 
         // METHODS
         private void NewFileMethod(object o)
         {
-            throw new System.NotImplementedException();
+            canvas.Clear();
+            OnPropertyChanged(new PropertyChangedEventArgs("dataChanged"));
+        }
+        private void OpenFileMethod(object o)
+        {
+            try
+            {
+                if (dialogService.OpenFileDialog()) 
+                {
+                    currentFileName = dialogService.FilePath;
+                    fileService.Load(canvas, currentFileName);
+                    dialogService.ShowMessage("File is opened and data is loaded from it.");
+                }
+            }
+            catch (System.Exception ex)
+            {
+                dialogService.ShowMessage(ex.Message);
+            }
+        }
+        private void SaveFileMethod(object o)
+        {
+            try
+            {
+                if (currentFileName.Contains("Pentagon Editor"))  

```

Comment:
"contains" is bad idea, path can easily contain these words, if so you will have unexpected behaviour. much easiear to have variable currentFileName to be "null" from the start and chacking for that

Reply:
okay